machine-learning/tutorials/TransferLearningTF/Program.cs

```diff
@@ -0,0 +1,182 @@
+            // Create and train the model based on the dataset that has been loaded, transformed.
+            // <SnippetTrainModel>
+            ITransformer model = estimator.Fit(data);
+            // </SnippetTrainModel>
+
+            // Process the training data through the model
+            // This is an optional step, but it's useful for debugging issues
+            // <SnippetTransformData>
+            var predictions = model.Transform(data);
+            // </SnippetTransformData>
+
+            // <SnippetEnumerateDataViews>
+            var imageNetData = mlContext.CreateEnumerable<ImageNetData>(data, false, true);
+            var imageNetPredictionData = mlContext.CreateEnumerable<ImageNetPrediction>(predictions, false, true);
+            // </SnippetEnumerateDataViews>
+
+            // <SnippetCallPairAndDisplayResults1>
+            PairAndDisplayResults(imageNetData, imageNetPredictionData);
+            // </SnippetCallPairAndDisplayResults1>
+
+            // Get some performance metrics on the model using training data
+            Console.WriteLine("=============== Classification metrics ===============");
+
+            // <SnippetEvaluate>           
+            var regressionContext = new MulticlassClassificationCatalog(mlContext);
+            var metrics = regressionContext.Evaluate(predictions, label: LabelTokey, predictedLabel: DefaultColumnNames.PredictedLabel);
+            // </SnippetEvaluate>
+
+            //<SnippetDisplayMetrics>
+            Console.WriteLine($"LogLoss is: {metrics.LogLoss}");
+            Console.WriteLine($"PerClassLogLoss is: {String.Join(" , ", metrics.PerClassLogLoss.Select(c => c.ToString()))}");
+            //</SnippetDisplayMetrics>
+
+            // Save the model to assets/outputs
+            Console.WriteLine("=============== Save model to local file ===============");
+
+            // <SnippetSaveModel>
+            using (var fileStream = new FileStream(outputModelLocation, FileMode.Create))
+                mlContext.Model.Save(model, fileStream);
+            // </SnippetSaveModel>
+
+            Console.WriteLine($"Model saved: {outputModelLocation}");
+        }
+
+        public static void ClassifyImages(MLContext mlContext, string dataLocation, string imagesFolder, string outputModelLocation)
+        {
+            Console.WriteLine($"=============== Loading model ===============");
+            Console.WriteLine($"Model loaded: {outputModelLocation}");
+
+            // Load the model
+            // <SnippetLoadModel>
+            ITransformer loadedModel;
+            using (var fileStream = new FileStream(outputModelLocation, FileMode.Open))
+                loadedModel = mlContext.Model.Load(fileStream);
+            // </SnippetLoadModel>
+
+            // <SnippetReadFromCSV> 
+            var imageNetData = ImageNetData.ReadFromCsv(dataLocation, imagesFolder);
```

Comment:
This is inconsistent with other snippets where you want to convert IDataView to an IEnumerable<T>. I would instead use `LoadFromTextFile<ImageNetData>` and then `CreateEnumerable<ImageNetData>`.

Reply:
We need read from csv to add the file path to the imagepath field